### PR TITLE
Add TwoPaneRight Layout

### LINKED ISSUE #1216: add Two Pane Right layout
**Is your feature request related to a problem? Please describe.**
I really like the two pane layout, but I'd like to have the main window to be on the right and cycle thru the other windows on the left.
What I am looking for is basically similar to the `Tall Right` layout.

**Describe the solution you'd like**
Basically the addition of said layout. I have seen some work been done on the ability to add custom layouts, but this request is rather basic as its just flipped. I might implement it if I ever figure out how to build this software myself.

**Describe alternatives you've considered**
To wait for the before mentioned custom layout feature.

### PR DESCRIPTION
**Pull request by TypicalFence**
Closes https://github.com/ianyh/Amethyst/issues/1216

I'd like to add it as a native layout because custom layouts do not support the native shrink/expand commands.

Getting xcode to work was quite the hassle, I've used an older version of xcode, but will update for future PR's.
I hope this is somehow mergeable as the xcode project files are confusing to me.

The changes to the podfile were needed for some reason because it wouldn't install the linter otherwise. Can't tell if this is a me problem or a general one. So maybe those changes should be removed?

### CODE REVIEW

File: Amethyst.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
Comment (rgeraskin) [inline Amethyst.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: looks like it's extra file that should not be committed

File: Amethyst.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (rgeraskin) [inline Amethyst.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: looks like it's extra file that should not be committed

File: Amethyst.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (rgeraskin) [inline Amethyst.xcworkspace/xcshareddata/swiftpm/Package.resolved:1]: looks like it's extra file that should not be committed

File: Amethyst/Debug/WindowsInfo.swift [hunk 1]
```diff
@@ -41,11 +41,13 @@ struct Windows: ParsableCommand {
         for application in applications where includeUnmanaged || application.isManageable == .manageable {
             let app = SIApplication(runningApplication: application)
             print("\(app.title() ?? "<no title>") (pid \(app.pid()))")
+            /*
```
Comment (rgeraskin) [inline Amethyst/Debug/WindowsInfo.swift:44]: why do we need to comment it in this PR?

File: Amethyst/Layout/Layouts/TwoPaneRightLayout.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        return windows.reduce([]) { acc, window -> [FrameAssignmentOperation<Window>] in
+            var assignments = acc
+            var windowFrame = CGRect.zero
+            let isMain = acc.count < mainPaneCount
+            var scaleFactor: CGFloat
+
+            if isMain {
+                scaleFactor = screenFrame.size.width / mainPaneWindowWidth
+                windowFrame.origin.x = screenFrame.origin.x + (isHorizontal ? mainPaneWindowWidth : 0)
+                windowFrame.origin.y = screenFrame.origin.y + (isHorizontal ? 0 : mainPaneWindowHeight)
```
Comment (rgeraskin) [inline Amethyst/Layout/Layouts/TwoPaneRightLayout.swift:79]: as this layout is kind of modified TwoPaneLayout with another x-axis behavior, do we need `windowFrame.origin.y` modifying?

File: Amethyst/Layout/Layouts/TwoPaneRightLayout.swift [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                windowFrame.size.width = mainPaneWindowWidth
+                windowFrame.size.height = mainPaneWindowHeight
+            } else {
+                scaleFactor = screenFrame.size.width / secondaryPaneWindowWidth
+                windowFrame.origin.x = screenFrame.origin.x
+                windowFrame.origin.y = screenFrame.origin.y
```
Comment (rgeraskin) [inline Amethyst/Layout/Layouts/TwoPaneRightLayout.swift:85]: as this layout is kind of modified TwoPaneLayout with another x-axis behavior, do we need `windowFrame.origin.y` modifying?

File: Amethyst/Layout/Layouts/TwoPaneRightLayout.swift [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+        return windows.reduce([]) { acc, window -> [FrameAssignmentOperation<Window>] in
+            var assignments = acc
+            var windowFrame = CGRect.zero
+            let isMain = acc.count < mainPaneCount
+            var scaleFactor: CGFloat
+
+            if isMain {
+                scaleFactor = screenFrame.size.width / mainPaneWindowWidth
+                windowFrame.origin.x = screenFrame.origin.x + (isHorizontal ? mainPaneWindowWidth : 0)
```
Comment (rgeraskin) [inline Amethyst/Layout/Layouts/TwoPaneRightLayout.swift:78]: We should add `secondaryPaneWindowWidth` value to get a proper x-position for main window

```
windowFrame.origin.x = screenFrame.origin.x + (isHorizontal ? secondaryPaneWindowWidth : 0)
```

File: Podfile
Comment (rgeraskin) [inline Podfile:1]: Unnecessary change

File: Podfile.lock
Comment (rgeraskin) [inline Podfile.lock:1]: Unnecessary change

**Comment by rgeraskin** [comment]
Hi! I've commented some doubtful changes. Also, I've successfully built this branch
1. With the latest XCode
2. With changes from my comments
3. Rebased on the current `development` branch

Works like a charm! Thank you, hopefully it will be merged in the upstream🙂

btw: I'm a total noob in swift, xcode and Amethyst codebase, be careful

**Comment by TypicalFence** [comment]
I don't use amethyst anymore and will not really merge it, I would suggest @rgeraskin creates their own PR.
